source/includes/connection-snippets/x509-snippets/c-connection.c

```diff
@@ -0,0 +1,30 @@
+// begin x509 connection
+#include <mongoc/mongoc.h>
+
+int
+main (int argc, char *argv[])
+{
+   mongoc_database_t *database;
+   mongoc_client_t *client;
+
+   mongoc_init ();
+
+   const mongoc_ssl_opt_t *ssl_default = mongoc_ssl_opt_get_default ();
+   mongoc_ssl_opt_t ssl_opts = { 0 };
+
+   memcpy(&ssl_opts, ssl_default, sizeof ssl_opts);
+
+   ssl_opts.pem_file = "/etc/certs/mongodb/client.pem";
+   ssl_opts.ca_file = "/etc/certs/mongodb/ca.pem";
+   ssl_opts.weak_cert_validation = false;
+   ssl_opts.allow_invalid_hostname = false;
```

Comment:
These `false` assignments are redundant because the `ssl_opts` struct was initialized with zero values for all fields (i.e. `{ 0 }` initializing struct).